### PR TITLE
 fill as many bottles as possible instead of giving up

### PR DESCRIPTION
**Pull request by adam-currie**
Currently if you sneak to fill all your bottles it won't let you fill any if you don't have enough exp or vault/economy money. So this pr makes it instead fill as many bottles as possible. If they don't even have enough for one, then it tells them and gives up basically like before.

I also got rid of the build directory and added an option to print a vault/currency receipt but those are separate commits, in case you don't like that for some reason.

I tested every change i made but i didn't know how to kind of prove that...

Btw nice plugin and nice code.

### CODE REVIEW

File: src/main/java/dev/cerus/xpbottler/XpBottlerPlugin.java [hunk 1]
```diff
@@ -73,42 +74,60 @@ public void onInteract(final PlayerInteractEvent event) {
                     return;
                 }
 
+                // Assumes player always has at least one bottle because we already checked that they are holding one
+                int bottlesToFill = player.isSneaking() ?
+                    Math.min(
+                        XpBottlerPlugin.this.getPlayerExp(player) / xpPerBottle, 
+                        item.getAmount())
+                    : (XpBottlerPlugin.this.getPlayerExp(player) >= xpPerBottle) ? 1 : 0;
```
Comment (cerus) [inline src/main/java/dev/cerus/xpbottler/XpBottlerPlugin.java:82]: I'm not a big fan of nested ternaries, but this is really minor so it doesn't really matter if you change it or leave it as is. :)
```suggestion
                int bottlesToFill;
                if(player.isSneaking()) {
                    bottlesToFill = Math.min(
                        XpBottlerPlugin.this.getPlayerExp(player) / xpPerBottle, 
                        item.getAmount()
                    );
                } else {
                    bottlesToFill = (XpBottlerPlugin.this.getPlayerExp(player) >= xpPerBottle) ? 1 : 0;
                }
```

**Review by cerus — APPROVED** [review]

**Comment by cerus** [comment]
Thanks for the contribution! Let me know if your PR is ready to merge

**Comment by adam-currie** [comment]
> Thanks for the contribution! Let me know if your PR is ready to merge

yep all good!

**Comment by cerus** [comment]
Thanks again!

**Pull request merged**